test/e2e/specs/site-editor/font-library.spec.js

```diff
@@ -15,15 +28,88 @@ test.describe( 'Font Library', () => {
 		} );
 
 		test( 'should display the "Manage Fonts" icon', async ( { page } ) => {
-			await page.getByRole( 'button', { name: /styles/i } ).click();
+			await page.getByRole( 'button', { name: 'Styles' } ).click();
 			await page
-				.getByRole( 'button', { name: /typography styles/i } )
+				.getByRole( 'button', { name: 'Typography Styles' } )
 				.click();
 			const manageFontsIcon = page.getByRole( 'button', {
-				name: /manage fonts/i,
+				name: 'Manage Fonts',
 			} );
 			await expect( manageFontsIcon ).toBeVisible();
 		} );
+
+		test( 'should allow user to upload multiple local font files', async ( {
+			page,
+			editor,
+		} ) => {
+			await page.getByRole( 'button', { name: 'Styles' } ).click();
+			await page
+				.getByRole( 'button', { name: 'Typography Styles' } )
+				.click();
+			await page
+				.getByRole( 'button', {
+					name: 'Manage Fonts',
+				} )
+				.click();
+
+			// Upload local fonts.
+			await page.getByRole( 'tab', { name: 'Upload' } ).click();
+			const fileChooserPromise = page.waitForEvent( 'filechooser' );
+			await page.getByRole( 'button', { name: 'Upload Font' } ).click();
+			const fileChooser = await fileChooserPromise;
+			// Provides coverage for https://github.com/WordPress/gutenberg/issues/59023.
+			await fileChooser.setFiles( [
+				'./test/e2e/assets/Exo2-Regular.woff',
+				'./test/e2e/assets/Exo2-SemiBoldItalic.woff2',
+			] );
+
+			// Check fonts were installed.
+			await expect(
+				page
+					.getByLabel( 'Upload' )
+					.getByText( 'Fonts were installed successfully.' )
+			).toBeVisible();
+			await page.getByRole( 'tab', { name: 'Library' } ).click();
+			// Provides coverage for https://github.com/WordPress/gutenberg/issues/60040.
+			await page.getByRole( 'button', { name: 'Exo 2' } ).click();
+			await expect( page.getByLabel( 'Exo 2 Normal' ) ).toBeVisible();
+			await expect(
+				page.getByLabel( 'Exo 2 Semi-bold Italic' )
+			).toBeVisible();
+
+			// Check CSS preset was created.
+			await page.getByRole( 'button', { name: 'Close' } ).click();
+			await page
+				.getByRole( 'button', { name: 'Typography Headings styles' } )
+				.click();
+			await page.getByLabel( 'Font' ).selectOption( 'Exo 2' );
+			await expect(
+				editor.canvas.locator( '.is-root-container h1' )
+			).toHaveCSS( 'font-family', '"Exo 2"' );
+		} );
+
+		test( 'should allow user to delete installed fonts', async ( {
+			page,
+		} ) => {
+			await page.getByRole( 'button', { name: 'Styles' } ).click();
+			await page
+				.getByRole( 'button', { name: 'Typography Styles' } )
+				.click();
+			await page
+				.getByRole( 'button', {
+					name: 'Manage Fonts',
+				} )
+				.click();
+
+			await page.getByRole( 'button', { name: 'Exo 2' } ).click();
```

Comment:
Agreed, I've moved the delete test so that it happens within the same block as the upload. This means the test will cover uploading the fonts, checking that's successful, and then deletes them and checks that's successful: https://github.com/WordPress/gutenberg/pull/60221/commits/b81f7b78a10edfa986641aa71e62f149f4e55d57.